src/test/java/one/util/streamex/StreamExTest.java

```diff
@@ -461,7 +461,22 @@ public void testFlatArray() {
         data.put(3, null);
         assertEquals(asList("a", "b", "c", "d"), StreamEx.of(data.entrySet()).flatArray(Entry::getValue).toList());
     }
-    
+
+    @Test
+    public void testFlatOption() {
+        Function<Integer, Optional<String>> literalOf = num -> num == 1
+                ? Optional.of("one")
+                : Optional.empty();
+
+        List<Integer> original = asList(1, 2, 3, 4);
+        List<String> expected = asList("one");
+        List<String> actual = StreamEx.of(original)
```

Comment:
Better to generate tests with `TestHelpers.streamEx()` (see examples in this file). In this case parallel stream will also be tested as well as some other tricky cases.